### PR TITLE
feat: Add CacheProvider

### PR DESCRIPTION
**Pull request by bruce-glazier**
## Changes

Part of https://github.com/lifeomic/react-native-sdk/pull/421

### CODE REVIEW

**Review by sternetj — APPROVED** [review]

**Review by markdlv — APPROVED** [review]
We'll want to be careful not to store too many things, or too much of one thing, in here.

**Comment by bruce-glazier** [comment]
> We'll want to be careful not to store too many things, or too much of one thing, in here.

We have a set maximum of 5000 entries before existing entries start getting trashed.

**Review by swain — COMMENTED** [review]
@bruce-glazier I'm curious: what are you planning to use this cache for?

**Comment by bruce-glazier** [comment]
> @bruce-glazier I'm curious: what are you planning to use this cache for?

Initially used in https://github.com/lifeomic/react-native-sdk/pull/421 to cache user profile information to reduce the number of fetches that occur.

File: src/hooks/CacheProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import React, { createContext, useContext } from 'react';
+import { Cache } from 'react-native-cache';
+
+type CacheContextProps = {
+  cache?: Cache;
+};
+
+export const CacheContext = createContext<CacheContextProps>({
+  cache: undefined,
+});
```
Comment (swain) [inline src/hooks/CacheProvider.tsx:11]: IMO, exporting a raw context is a code smell. Do we need to export this?

**Comment by bruce-glazier** [comment]
Using persistQueryClient instead of this custom rolled cache provider

**Pull request closed without merging**